### DISCUSSION TITLE
Question about the specifications for interactive-presentation-backend-level-1

### DISCUSSION THREAD

**Discussion by IonelPopJara** [DevSkillsHQ/discusssions#10] [Help] [answered]
I was running the tests for the application and I noticed something.

The test first POST a presentation with two polls. And then it does a PUT and it's expecting the `Second Poll` as a result. Since the specifications don't say anything about it, I was wondering. Do I have to return them in the reverse order the polls were written? Or should the POST method also put the first presentation (this one doesn't make sense to me)?

The way I solved it, is by getting the polls in the reverse order they were written. It passed all the tests but doesn't feel like the intended solution.

**Comment by pronoooobster** [comment]
I have the same question.

**Comment by fenske** [comment, marked as answer]
Hi folks. Anton from DevSkills is here. 

This is sort of part of the main exercise objective, but the answer was pretty much listed in the original question. So I feel it's safe to elaborate. 😄 

The best way to answer this question is to try the hosted API attached in the spec.

https://user-images.githubusercontent.com/1162212/224841690-099976f9-1ba3-4e1d-9ee1-f835358208e8.mp4

As for why it works this way, I haven't authored this challenge, but there are a couple of ways to implement this API, and IMO, there isn't really a single correct approach. 

One is what @IonelPopJara suggested when creating a new presentation doesn't set the initial slide.

On the other hand, if we look at a real-life example with Google Slides. When you click on the link for a presentation, you immediately end up on Slide 1. 🤷 

Also, the fact that you managed to make the tests pass by reversing the order of the slides suggests we need to improve the E2E tests (e.g. before doing a PUT, do a GET to /current first and assert the expected response). 😅 

Hope I managed to clarify it for you.

Let me know if you have any other questions, and I wish you the best of luck with the rest of your interview processes!

Best,
Anton.

**Reply by IonelPopJara** [reply]
Thank you for the reply!

**Reply by pronoooobster** [reply]
Thanks, now it makes sense :smile:
